File: BugTracker.API/Models/Project.cs
using System;
using System.Collections.Generic;

namespace BugTracker.API.Models
{
    public class Project
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime DeadTime { get; set; }
        public bool isArchived { get; set; }
        public ICollection<Ticket> Tickets { get; set; }
        public ICollection<User> Users { get; set; }
    }
}

File: BugTracker-SPA/src/app/_models/Project.ts
import { Ticket } from './Ticket';
import { User } from './User';

export interface Project {
    id: number;
    title: string;
    description: string;
    startTime: Date;
    deadTime: Date;
    isArchived: boolean;
    ticketCount: number;

    tickets?: Ticket[];
    users?: User[];
}
